packages/cli/src/util/get-latest-version/index.ts

```diff
@@ -67,25 +69,26 @@ export default function getLatestVersion({
     }
   }
 
-  if (!cache || cache.expireAt < Date.now()) {
+  if (!cache || !cache.expireAt || cache.expireAt < Date.now()) {
     spawnWorker(
       {
         cacheFile,
         distTag,
-        updateCheckInterval,
         name: pkg.name,
+        updateCheckInterval,
       },
       output
     );
   }
 
   if (
     cache &&
-    !cache.notified &&
+    (!cache.notifyAt || cache.notifyAt < Date.now()) &&
     pkg.version &&
+    cache.version &&
     semver.lt(pkg.version, cache.version)
   ) {
-    cache.notified = true;
+    cache.notifyAt = Date.now() + notifyInterval;
     outputJSONSync(cacheFile, cache);
     return cache.version;
   }
```

Comment:
**suggestion (non-blocking):** simplify conditional expression

This conditional has a complex expression already. I think it will be easier to understand if we extract some named things, like:

```js
function isPkgOld(pkg, cache) {
  if (!pkg?.version || !cache?.version) {
    return false;
  }
  
  return semver.lt(pkg.version, cache.version);
}

const currentPkgIsOld = isPkgOld(pkg, cache);
const shouldNotify = !cache.notifyAt || Date.now() > cache.notifyAt;

if (currentPkgIsOld && shouldNotify) {
  cache.notifyAt = Date.now() + notifyInterval;
  outputJSONSync(cacheFile, cache);
  return cache.version;
}
```

Reply:
Cool. I simplified it without needing to do a function.